crates/router/src/core/user_role/role.rs

```diff
@@ -207,14 +199,10 @@ pub async fn update_role(
         utils::user_role::validate_role_groups(groups)?;
     }
 
-    let role_info = roles::RoleInfo::from_role_id_in_merchant_scope(
-        &state,
-        role_id,
-        &user_from_token.merchant_id,
-        &user_from_token.org_id,
-    )
-    .await
-    .to_not_found_response(UserErrors::InvalidRoleOperation)?;
+    let role_info =
+        roles::RoleInfo::from_role_id_and_org_id(&state, role_id, &user_from_token.org_id)
+            .await
+            .to_not_found_response(UserErrors::InvalidRoleOperation)?;
```

Comment:
We should also check if user has access to update this role. This can be done by getting the role using the lineage.